sdks/functions.do/README.md

```diff
@@ -10,17 +10,103 @@ Key challenges include:
 - **Prompt Engineering**: More art than science, difficult to standardize
 - **Configuration**: Complex parameter interactions require careful tuning
 
-### The Solution
+## The Solution
 
 functions.do creates a clean separation between AI capabilities and application code through strongly-typed interfaces that hide model complexities, enabling:
 
 - Rapid prototyping of AI applications
 - Continuous improvement without disrupting application code
 - Comprehensive evaluation and optimization strategies
 
-### Usage
+## Installation
 
-You can use it by simply calling any function name with any arguments on the `ai` object like:
+```bash
+npm install functions.do
+# or
+yarn add functions.do
+# or
+pnpm add functions.do
+```
+
+## API Overview
+
+The functions.do SDK exports two main components:
+
+- `ai`: A flexible proxy for invoking AI functions with various patterns
+- `AI`: A function for defining schemas and creating custom AI functions
+
+## Usage Examples
+
+### Using the `ai` Proxy
+
+The `ai` export is a versatile proxy that supports multiple invocation patterns:
+
+#### Tagged Template Literals
+
+```typescript
+import { ai } from 'functions.do'
+
+// Basic usage
+const result = await ai`Summarize this article: ${articleText}`
+
+// With configuration
+const configuredResult = await ai({ model: 'grok-3' })`Translate this to French: ${text}`
+```
+
+#### Function Invocation
+
+```typescript
+import { ai } from 'functions.do'
+
+// Simple function call without schema
+const storyBrand = await ai.storyBrand({ company: 'Vercel' })
+
+// Function call with schema (structured output)
+const leanCanvas = await ai.leanCanvas({ company: 'Cloudflare' }, {
+  productName: 'name of the product or service',
+  problem: ['top 3 problems the product solves'],
+  solution: ['top 3 solutions the product offers'],
+  uniqueValueProposition: 'clear message that states the benefit of your product',
+  unfairAdvantage: 'something that cannot be easily copied or bought',
+  customerSegments: ['list of target customer segments'],
+  keyMetrics: ['list of key numbers that tell you how your business is doing'],
+  channels: ['path to customers'],
+  costStructure: ['list of operational costs'],
+  revenueStreams: ['list of revenue sources'],
+  recommendations: ['list of recommendations based on the analysis'],
+})
+
+// Function call with schema and AI configuration
+const blogTitles = await ai.listBlogPostTitles(
+  { topic: 'the future of work post-API' },
+  ['list SEO-optimized titles'],
+  { model: 'gpt-4.5' }
+)
+```
+
+### Using the `AI` Function
+
+The `AI` function allows you to define custom AI functions with specific schemas:
+
+```typescript
+import { AI } from 'functions.do'
+
+// Define a custom AI function with configuration
+export default AI({
+  defineFunction: {
+    _model: 'claude-3.7-sonnet', // config props are pre-pended with `_`
```

Comment:
Let's replace this example with a better one ... how about a book proposal one like in the examples folder